components/core/src/clp/Grep.cpp

```diff
@@ -540,76 +557,346 @@ std::optional<Query> Grep::process_raw_query(
+                        for (char const& c : current_string) {
+                            if (c == '*') {
+                                contains_wildcard = true;
+                                regex_search_string.push_back('.');
+                                if (pos > 0 && pos < current_string.size() - 1) {
+                                    contains_central_wildcard = true;
+                                }
+                            } else if (log_surgeon::SchemaParser::get_special_regex_characters()
+                                               .find(c)
+                                       != log_surgeon::SchemaParser::get_special_regex_characters()
+                                                  .end())
+                            {
+                                regex_search_string.push_back('\\');
+                            }
+                            regex_search_string.push_back(c);
+                            pos++;
+                        }
+                        log_surgeon::NonTerminal::m_next_children_start = 0;
+                        log_surgeon::Schema schema2;
+                        // TODO: we don't always need to do a DFA intersect
+                        //       most of the time we can just use the forward
+                        //       and reverse lexers which is much much faster
+                        // TODO: NFA creation not optimized at all
+                        schema2.add_variable("search", regex_search_string, -1);
+                        RegexNFA<RegexNFAByteState> nfa;
+                        std::unique_ptr<SchemaAST> schema_ast = schema2.release_schema_ast_ptr();
+                        for (std::unique_ptr<ParserAST> const& parser_ast :
+                             schema_ast->m_schema_vars)
+                        {
+                            auto* schema_var_ast = dynamic_cast<SchemaVarAST*>(parser_ast.get());
+                            ByteLexer::Rule rule(0, std::move(schema_var_ast->m_regex_ptr));
+                            rule.add_ast(&nfa);
+                        }
+                        // TODO: DFA creation isn't optimized for performance
+                        //       at all
+                        // TODO: log-surgeon code needs to be refactored to
+                        //       allow direct usage of DFA/NFA without lexer
+                        unique_ptr<RegexDFA<RegexDFAByteState>> dfa2
+                                = forward_lexer.nfa_to_dfa(nfa);
+                        unique_ptr<RegexDFA<RegexDFAByteState>> const& dfa1
+                                = forward_lexer.get_dfa();
+                        schema_types = dfa1->get_intersect(dfa2);
+                        bool already_added_var = false;
+                        for (int id : schema_types) {
+                            auto& schema_type = forward_lexer.m_id_symbol[id];
+                            if (schema_type != "int" && schema_type != "float") {
```

Comment:
Instead of using magic strings, is there a way we can get them from the schema instead?

Reply:
Its kind of awkward for the schema to know which vars are encoded because the user doesn't tell it (and probably shouldn't?). Log-Surgeon doesn't deal with encoding at all. Right now CLP just hopes the user will implement a schema that specifies int and float correctly, and then tries to encode them with a failsafe of treating them as dictionary variables if it fails. Potentially we could let the user specify if CLP should encode int/float/etc.. If that option is turned on for int, for example, then int would not be allowed to be specified in the user schema, but instead would be added to the schema by CLP itself using a hardcoded int regex.